storage/innobase/buf/buf0flu.cc

```diff
@@ -112,11 +112,17 @@ static void buf_flush_validate_skip()
 }
 #endif /* UNIV_DEBUG */
 
-void buf_pool_t::page_cleaner_wakeup(bool for_LRU)
+void buf_pool_t::page_cleaner_wakeup(wakeup_reason reason)
 {
   if (!page_cleaner_idle())
   {
-    if (for_LRU)
+    /* Many ISA provide N and Z flags, which we desire to use for
+    efficient checking of the wakeup reason. */
+    static_assert(WAKE_NOW_LRU < 0, "efficiency");
+    static_assert(WAKE_IDLE == 0, "efficiency");
+    static_assert(WAKE_IDLE_LRU > 0, "efficiency");
+
+    if (reason < WAKE_IDLE)
       /* Ensure that the page cleaner is not in a timed wait. */
       pthread_cond_signal(&do_flush_list);
     return;
```

Comment:
For the record, retaining the equivalent condition `reason != WAKE_IDLE` would result in roughly the same maximum latency in my 160-thread `oltp_read_only` test on 50M buffer pool, 1.2G data, 100M log, run on RAM disk. The maximum latency would be about 48ms.

That is, this change of the parameter from `bool` to `enum` might not be useful.